server/src/main/java/org/elasticsearch/action/bulk/TransportBulkAction.java

```diff
@@ -388,7 +388,7 @@ private long buildTookInMillis(long startTimeNanos) {
      * */
     private final class BulkOperation extends ActionRunnable<BulkResponse> {
         private final Task task;
-        private final BulkRequest bulkRequest;
+        private volatile BulkRequest bulkRequest; // set to null once all requests are sent out
```

Comment:
NIT: no need for this to be `volatile` I think, you're only ever touching it on the thread that nulls it out?